app/controllers/descriptives_controller.rb

```diff
@@ -39,16 +39,15 @@ def display_error(error)
   end
 
   def convert_metdata_success(description:)
-    valid_model = begin
-      @cocina.new(description: description)
-    rescue Cocina::Models::ValidationError => e
-      return display_error(e.message) # rubocop:disable Lint/NoReturnInBeginEndBlocks
-    end
-
-    Repository.store(valid_model)
-    redirect_to solr_document_path(@cocina.externalIdentifier),
-                status: :see_other,
-                notice: 'Descriptive metadata has been updated.'
+    CocinaValidator.validate(@cocina, description: description).either(
```

Comment:
For what it is worth, I find this syntax obtuse compared against just evaluating the Success/Failure result.